search-impl/src/main/java/com/example/auction/search/impl/BrokerEventConsumer.java

```diff
@@ -0,0 +1,70 @@
+package com.example.auction.search.impl;
+
+import akka.stream.javadsl.Flow;
+import com.example.auction.bidding.api.BidEvent;
+import com.example.auction.bidding.api.BiddingService;
+import com.example.auction.item.api.ItemEvent;
+import com.example.auction.item.api.ItemService;
+import com.example.auction.search.IndexedStore;
+import com.example.elasticsearch.IndexedItem;
+import com.lightbend.lagom.javadsl.api.broker.Topic;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+
+@Singleton
+public class BrokerEventConsumer {
+
+    private IndexedStore indexedStore;
+    private ItemService itemService;
+    private BiddingService biddingService;
+
+    @Inject
+    public BrokerEventConsumer(IndexedStore indexedStore, ItemService itemService, BiddingService biddingService) {
```

Comment:
And here's the Class taking care of event consumption. It is now doing both topics in a single class. I don't want to strip it further because it'd take the abstraction too far.